include/nbl/builtin/hlsl/cpp_compat/intrinsics.hlsl

```diff
@@ -151,8 +154,8 @@ inline T max(NBL_CONST_REF_ARG(T) a, NBL_CONST_REF_ARG(T) b)
 #endif
 }
 
-template<typename FloatingPoint>
-inline FloatingPoint isnan(NBL_CONST_REF_ARG(FloatingPoint) val)
+template<typename FloatingPoint NBL_FUNC_REQUIRES(hlsl::is_floating_point_v<FloatingPoint>)
+inline bool isnan(NBL_CONST_REF_ARG(FloatingPoint) val)
```

Comment:
`is_floating_point` matches both `float` and `vector<float` correct?

we should really move inf, nan detectors to `tgmath` and in the `tgmath` have a general channel/vector forwarder which for the C++ function calls can perform pow, exp2, etc. on vectors (because spirv already can!)

Reply:
> `is_floating_point` matches both `float` and `vector<float` correct?

correct, `is_floating_point` derives from https://github.com/Devsh-Graphics-Programming/Nabla/blob/master/include/nbl/builtin/hlsl/type_traits.hlsl#L166

Reply:
>we should really move inf, nan detectors to tgmath and in the tgmath have a general channel/vector forwarder which for the C++ function calls can perform pow, exp2, etc. on vectors (because spirv already can!)

TODO